### PR TITLE
feat(organisms): add SfTable component

### LINKED ISSUE #513: [Mobile/Desktop] Implement SfTable
Screenshot from 2019-12-06 15-28-13
Screenshot from 2019-12-06 15-28-21

Can be found in desgins in My Account

Proposed API:
```vue
<SfTable>
  <SfTableRow>
    <SfTableHeader><!-- slot content --><SfTableHeader>
    <SfTableHeader><!-- slot content --><SfTableHeader>
  </SfTableRow>
  <SfTableRow>
    <SfTableData><!-- slot content --><SfTableData>
    <SfTableData><!-- slot content -->.<SfTableData>
    <SfTableData><!-- slot content --><SfTableData>
  </SfTableRow>
</SfTable>
```

### PR DESCRIPTION
**Pull request by aniamusial**
# Related issue
Closes #513 

# Scope of work
Implemented SfTable component

# Checklist

- [x] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/shared/styles/components/SfTable.scss [hunk 1]
```diff
@@ -0,0 +1,72 @@
+@import "../variables";
+
+:root {
+  --table-column-count: 6;
```
Comment (przspa) [inline packages/shared/styles/components/SfTable.scss]: why 6? why not 1, 4 or 10?

File: packages/shared/styles/components/SfTable.scss [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+.sf-table {
+  width: 100%;
+  border: none;
+  color: $c-text-muted;
```
Comment (przspa) [inline packages/shared/styles/components/SfTable.scss]:
```suggestion
```
we should style just table elements

File: packages/shared/styles/components/SfTable.scss [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+.sf-table-data {
+  text-align: center;
```
Comment (przspa) [inline packages/shared/styles/components/SfTable.scss]:
```suggestion
```

File: packages/vue/src/components/organisms/SfTable/SfTable.html [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<table class="sf-table" cellspacing="0" cellpadding="0">
```
Comment (przspa) [inline packages/vue/src/components/organisms/SfTable/SfTable.html]:
```suggestion
<table class="sf-table" cellspacing="0" cellpadding="0">
```
```suggestion
<table class="sf-table">
```
use css for do that

File: packages/shared/styles/components/SfTable.scss (hunk 1 above)
Comment (przspa) [reply packages/shared/styles/components/SfTable.scss]: :root, it should be local variable for sf-table :)

File: packages/shared/styles/components/SfTable.scss [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+}
+
+.sf-table-row {
+  display: flex;
+  flex-wrap: wrap;
+  border-bottom: 1px solid $c-light;
+
+  th {
```
Comment (przspa) [inline packages/shared/styles/components/SfTable.scss]: th, td? we should use BEM for that 🙏

Comment (przspa) [reply packages/shared/styles/components/SfTable.scss]: `sf-table-data`, `sf-table-data--td` `sf-table-data--th`

File: packages/shared/styles/components/SfTable.scss [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  td {
+    &:nth-child(odd) {
+      font-weight: 500;
+    }
+
+    @media (min-width: $desktop-min) {
+      &:nth-child(odd) {
+        font-weight: $body-font-weight-primary;
```
Comment (przspa) [inline packages/shared/styles/components/SfTable.scss]:
```suggestion
```

File: packages/vue/src/components/organisms/SfTable/SfTable.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    };
+  },
+  components: {
+    SfButton,
+    SfTableRow,
+    SfTableData
+  },
+  mounted() {
+    this.columnsCount = this.$slots.default[0].componentInstance.$children.length;
+    document
```
Comment (przspa) [inline packages/vue/src/components/organisms/SfTable/SfTable.js]: you should use set it for sf-table

File: packages/vue/src/components/organisms/SfTable/SfTable.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+      columnsCount: 0
+    };
+  },
+  components: {
+    SfButton,
+    SfTableRow,
+    SfTableData
+  },
+  mounted() {
+    this.columnsCount = this.$slots.default[0].componentInstance.$children.length;
```
Comment (przspa) [inline packages/vue/src/components/organisms/SfTable/SfTable.js]: that's not good idea

File: packages/shared/styles/components/SfTable.scss [hunk 9]
```diff
@@ -0,0 +1,46 @@
+@import "../variables";
+
+.sf-table {
+  width: 100%;
+  padding: 0;
+  border-spacing: 0;
+  border: none;
+  &-row {
```
Comment (filrak) [inline packages/shared/styles/components/SfTable.scss]: in BEM it should be `--row` ;)

File: packages/shared/styles/components/SfTable.scss [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+      color: $c-text;
+
+      @media (min-width: $desktop-min) {
+        border-bottom: 1px solid $c-light;
+        flex-wrap: nowrap;
+        background-color: $c-white;
+      }
+    }
+  }
+  &-data {
```
Comment (filrak) [inline packages/shared/styles/components/SfTable.scss]: same, not very BEMish

File: packages/vue/public/assets/storybook/download.svg [hunk 11]
```diff
@@ -0,0 +1,4 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (filrak) [inline packages/vue/public/assets/storybook/download.svg]: we can add this to `icons.js`

File: packages/vue/src/components/organisms/SfTable/SfTable.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
+import SfTableRow from "./_internal/SfTableRow.vue";
+import SfTableData from "./_internal/SfTableData.vue";
+import SfButton from "../../atoms/SfButton/SfButton.vue";
+
+import Vue from "vue";
+
+Vue.component("SfTableRow", SfTableRow);
+Vue.component("SfTableData", SfTableData);
```
Comment (filrak) [inline packages/vue/src/components/organisms/SfTable/SfTable.js]: Why not `SfTableColumn` ?

File: packages/vue/src/components/organisms/SfTable/SfTable.stories.js [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+          ["#35767", "4th Nov", "Paypal", "12.00$"],
+          ["#35767", "4th Nov", "Visa", "15.00$"],
+          ["#35767", "4th Nov", "Paypal", "12.00$"]
+        ]
+      };
+    },
+    components: { SfTable, SfButton },
+    template: `<SfTable>
+    <template>
+      <SfTableRow class="sf-table-row--header">
```
Comment (filrak) [inline packages/vue/src/components/organisms/SfTable/SfTable.stories.js]: Wouldn't it be wiser to have separate `SfTableHeader` component? `is-header` prop and adidng additional CSS rules doesn't seem like a good developer experience

File: packages/vue/src/components/organisms/SfTable/_internal/SfTableRow.js [hunk 14]
```diff
@@ -0,0 +1,8 @@
+export default {
+  name: "SfTableRow",
+  mounted() {
+    this.$parent.updateColumnsCount(
```
Comment (filrak) [inline packages/vue/src/components/organisms/SfTable/_internal/SfTableRow.js]: sometimes people might want to wrap columns into links etc, better do a check if there is parent and catch error OR use provide/inject

Comment (przspa) [reply packages/vue/src/components/organisms/SfTable/_internal/SfTableRow.js]: wrap td / th into links it's a semantic way?

Comment (przspa) [reply packages/vue/src/components/organisms/SfTable/_internal/SfTableRow.js]: but sure :) provide/inject is ok 👍

Comment (filrak) [reply packages/vue/src/components/organisms/SfTable/_internal/SfTableRow.js]: @przspa expect unexpected, here people might not be aware that its semantically correct table ;)

**Pull request merged**